src/asm/symbol.c

```diff
@@ -103,7 +103,7 @@ static char const *Callback__FILE__(void)
 	size_t j = 1;
 
 	/* TODO: is there a way for a file name to be empty? */
-	assert(strlen(fileName) != 0);
+	assert(fileName[0]);
```

Comment:
(This *is* terser, but for what it's worth, [gcc](https://godbolt.org/z/rW6z1s) and [clang](https://godbolt.org/z/cjMsv3) will optimize both the same way, even at `-O1`.)

Reply:
Yes, but as mentioned in the commit description, Clang's static analyzer fails to reduce the former to the latter, and thus generates a bogus bug report.